core/src/main/java/com/yahoo/ycsb/generator/graph/randomcomponents/RandomGraphComponentGenerator.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2018 YCSB contributors. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.generator.graph.randomcomponents;
+
+import com.yahoo.ycsb.generator.StoringGenerator;
+import com.yahoo.ycsb.generator.graph.Edge;
+import com.yahoo.ycsb.generator.graph.GraphComponent;
+import com.yahoo.ycsb.generator.graph.GraphDataGenerator;
+import com.yahoo.ycsb.generator.graph.Node;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Abstract class to pick a random {@link GraphComponent} ({@link Node} or {@link Edge}).
+ */
+public abstract class RandomGraphComponentGenerator extends StoringGenerator<GraphComponent> {
+
+  private static final String NODE_FILE_NAME = "nodeIds.txt";
+  private static final String EDGE_FILE_NAME = "edgeIds.txt";
+  private static final String COMPONENT_FILE_NAME = "componentIds.txt";
+  private static final String CLASS_NAME = RandomGraphComponentGenerator.class.getSimpleName();
+
+  private final File nodeFile;
+  private final File edgeFile;
+  private final File componentFile;
+  private final GraphDataGenerator graphDataGenerator;
+  private GraphComponent lastGraphComponent;
+
+  RandomGraphComponentGenerator(String directory, GraphDataGenerator graphDataGenerator) throws IOException {
```

Comment:
Why restrict to IOException and then need to rethrow? I'd recommend no throws clause and throwing runtime exceptions, or throws Exception.